### PR TITLE
[SPARKR-233] [SPARKR-190] add more API for Column

### PR DESCRIPTION
**Pull request by davies**
Also replace "." as "_" in column names

### CODE REVIEW

**Comment by shivaram** [comment]
@davies Looks like there is an test failure here in the newly added test case

File: pkg/NAMESPACE [hunk 1]
```diff
@@ -127,10 +127,17 @@ exportClasses("Column")
 
 exportMethods("asc",
               "desc",
-              "first",
               "last",
               "lower",
               "upper",
+              "like",
```
Comment (shivaram) [inline pkg/NAMESPACE]: Probably not related to this PR, but could you sort this list ?

**Comment by shivaram** [comment]
Thanks for the change. I just had a minor comment, otherwise LGTM.

File: pkg/NAMESPACE (hunk 1 above)
Comment (davies) [inline pkg/NAMESPACE]: I'd like to it later, sort all of them.

Comment (shivaram) [inline pkg/NAMESPACE]: Ok - It'll be great if you could also sort the generics file (sorted per section)

**Comment by shivaram** [comment]
Merging this

**Pull request merged**